actionpack/lib/action_dispatch/http/cache.rb

```diff
@@ -196,7 +196,11 @@ def merge_and_normalize_cache_control!(cache_control)
           control.merge! cache_control
 
           if control[:no_store]
-            self._cache_control = NO_STORE
+            options = []
+            options << PRIVATE if control[:private]
+            options << NO_STORE
+
+            self._cache_control = options.join(", ")
```

Comment:
Since all conditions have `options` now, we can pull out the `options = []` and `self._cache_control = options.join(", ")` to before and after the `if` block.

Reply:
Good suggestion, done!